modules/standard/Sets.chpl

```diff
@@ -0,0 +1,496 @@
+    */
+    proc add(const ref x: eltType) {
+      return;
+    }
+
+    /*
+      Returns `true` if the given element is a member of this set, and `false`
+      otherwise.
+
+      :arg x: The element to test for membership.
+      :return: Whether or not the given element is a member of this set.
+    */
+    proc const contains(const ref x: eltType): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set shares no elements in common with the set
+      `other`, and `false` otherwise.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set and `other` are disjoint.
+    */
+    proc const disjoint(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set and `other` have at least one element in
+      common.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set and `other` intersect.
+    */
+    proc const intersect(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if the value of every element in this set is contained
+      in the set `other`.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a subset of `other`.
+    */
+    proc const subset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set is a proper subset of `other`. That is,
+      the value of every element in this set is contained in `other`, and
+      this set does not equal `other`.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a proper subset of `other`.
+    */
+    proc const properSubset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if the value of every element in the set `other` is
+      contained in this set.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a superset of `other`.
+    */
+    proc const superset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Returns `true` if this set is a proper superset of `other`. That is,
+      the value of every element in `other` is contained in this set, and
+      this set does not equal `other`.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set is a proper superset of `other`.
+    */
+    proc const properSuperset(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Remove the item from this set with a value equal to `x`.
+
+      .. warning::
+
+        Removing an element from this set may invalidate existing references
+        to the elements contained in this set.
+    
+      :arg x: The element to remove.
+
+      :throws IllegalArgumentError: If the list contains no such element.
+    */
+    proc remove(const ref x: eltType): eltType throws {
```

Comment:
Why does this return `eltType` ?

Reply:
If `remove` does not return a result, then there is no method capable of removing a element by value and then returning it.

I figure having such a method in the interface would be valuable. However this does conflate the semantics of the `set` remove with the `list` remove.

Reply:
OK